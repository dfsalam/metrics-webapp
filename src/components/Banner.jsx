import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Banner.css';

const Banner = ({ title, sub1, sub2 }) => (
  <div className="banner-container">
    <div className="banner-text">
      <h2>{title}</h2>
      <h3>{sub1}</h3>
      <h3>{sub2}</h3>
    </div>
  </div>
);

Banner.propTypes = {
  title: PropTypes.string.isRequired,
  sub1: PropTypes.string.isRequired,
  sub2: PropTypes.string,
};

Banner.defaultProps = {
  sub2: '',
};

export default Banner;
