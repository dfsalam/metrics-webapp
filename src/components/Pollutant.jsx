import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Pollutant.css';

const Pollutant = ({ name, value }) => {
  const output = `${value} Î¼g/m`;
  return (
    <div className="pol-container">
      <h3 className="pol-text">{name}</h3>
      <h3 className="pol-text">
        {output}
        <sup>3</sup>
      </h3>
    </div>
  );
};

Pollutant.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
};

export default Pollutant;
