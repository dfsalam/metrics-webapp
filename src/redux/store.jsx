import { configureStore } from '@reduxjs/toolkit';
import townsReducer from './towns/townsSlice';

const store = configureStore({
  reducer: {
    towns: townsReducer,
  },
});

export default store;
