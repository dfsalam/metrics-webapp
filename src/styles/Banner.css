body {
  -ms-overflow-style: none;
  scrollbar-width: none;
  background-color: rgb(237, 101, 149);
}

body::-webkit-scrollbar {
  display: none;
}

.banner-container {
  background-color: rgb(252, 81, 147);
  display: flex;
  background-image: url('../img/map.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: bottom right;
  min-height: 300px;
  justify-content: flex-end;
  align-items: flex-end;
}

.banner-text {
  width: 60%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-end;
  margin: 0 15px 15px 0;
}

.banner-text h2,
h3 {
  color: white;
  margin: 0;
  padding: 0;
  font-family: 'Gill Sans', sans-serif;
  font-weight: bold;
}
