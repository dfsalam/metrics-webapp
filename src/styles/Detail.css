.detail-container {
  background-color: rgb(225, 84, 150);  
}

.detail-container h2 {
  font-family: 'Gill Sans', serif;
  color: white;
  margin-left: 5px;
  padding-bottom: 0;
  margin-bottom: 5px;
  margin-top: 2px;
}

.detail-container h3 {
  font-family: 'Gill Sans', serif;
  color: white;
  margin-left: 5px;
  padding-bottom: 0;
  margin-bottom: 5px;
  margin-top: 2px;
}