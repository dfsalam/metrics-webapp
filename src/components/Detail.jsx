import React from 'react';
import PropTypes from 'prop-types';
import { useAppSelector } from '../redux/redux-hooks';
import '../styles/Detail.css';
import image from '../img/map.jpg';

const Detail = ({ town }) => {
  const { polution } = useAppSelector((state) => state.polution);
  if (polution.length === 0) {
    return (<h3>Is loading...</h3>);
  }
  const { lat, lon } = polution.coord;
  const {
    co, no, no2, o3, pm25, pm10, so2,
  } = polution.list[0].components;
  return (
    <div className="detail-container">
      <h2>{town}</h2>
      <div>
        <div>
          <h3>
            Latitud:
            {' '}
            {lat}
          </h3>
          <h3>
            Longitud:
            {' '}
            {lon}
          </h3>
        </div>
        <img alt="map" src={image} />
      </div>
      <h3>
        CO (Carbon monoxide),
        {' '}
        {co}
        {' '}
        μg/m3
      </h3>
      <h3>
        NO (Nitrogen monoxide),
        {' '}
        {no}
        {' '}
        μg/m3
      </h3>
      <h3>
        NO2 (Nitrogen dioxide),
        {' '}
        {no2}
        {' '}
        μg/m3
      </h3>
      <h3>
        O3 (Ozone),
        {' '}
        {o3}
        {' '}
        μg/m3
      </h3>
      <h3>
        SO2 (Sulphur dioxide),
        {' '}
        {no}
        {' '}
        μg/m3
      </h3>
      <h3>
        PM2.5 (Fine particles matter),
        {' '}
        {pm25}
        {' '}
        μg/m3
      </h3>
      <h3>
        PM10 (Coarse particulate matter),
        {' '}
        {pm10}
        {' '}
        μg/m3
      </h3>
      <h3>
        NH3 (Ammonia),
        {' '}
        {so2}
        {' '}
        μg/m3
      </h3>
    </div>
  );
};

Detail.propTypes = {
  town: PropTypes.string.isRequired,
};

export default Detail;
