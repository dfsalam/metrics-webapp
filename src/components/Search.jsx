import React from 'react';
import { useAppDispatch } from '../redux/redux-hooks';
import { getTowns, searchTown } from '../redux/towns/townsSlice';
import '../styles/Search.css';

const Search = () => {
  const dispatch = useAppDispatch();
  const inp = document.getElementById('searchInput');
  const handler = (desired) => {
    inp.addEventListener('keydown', (event) => {
      const { key } = event;
      if (key === 'Backspace') {
        dispatch(getTowns());
      }
    });
    dispatch(searchTown(desired));
  };

  return (
    <input type="text" id="searchInput" placeholder="Search..." onChange={(e) => handler(e.target.value)} />
  );
};

export default Search;
