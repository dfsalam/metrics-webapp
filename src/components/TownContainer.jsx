import React from 'react';
import Town from './Town';
import { useAppSelector } from '../redux/redux-hooks';
import '../styles/TownContainer.css';

const TownContainer = () => {
  const { towns, townsSearch, searching } = useAppSelector((state) => state.towns);

  if (searching) {
    return (
      <div>
        <h2 className="cntTitle">STATS BY TOWN</h2>
        <div className="cnt-body">
          {townsSearch.map((item) => (
            <Town
              key={item.townId}
              townId={item.townId}
              townName={item.townName}
            />
          ))}
        </div>
      </div>
    );
  }

  return (
    <div>
      <h2 className="cntTitle">STATS BY TOWN</h2>
      <div className="cnt-body">
        {towns.map((item) => (
          <Town
            key={item.townId}
            townId={item.townId}
            townName={item.townName}
          />
        ))}
      </div>
    </div>
  );
};

export default TownContainer;
