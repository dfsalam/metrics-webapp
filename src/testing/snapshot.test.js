import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import Detail from '../components/Detail';
import Search from '../components/Search';
import TownContainer from '../components/TownContainer';
import Banner from '../components/Banner';
import Pollutant from '../components/Pollutant';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('Components render correctly', () => {
  test('Detail', () => {
    const tree = render(
      <Detail />,
    );
    expect(tree).toMatchSnapshot();
  });
  test('Search', () => {
    const tree = render(<Search />);
    expect(tree).toMatchSnapshot();
  });
  test('TownContainer', () => {
    const tree = render(<TownContainer />);
    expect(tree).toMatchSnapshot();
  });
  test('Banner', () => {
    const tree = render(<Banner />);
    expect(tree).toMatchSnapshot();
  });
  test('Pollutant', () => {
    const tree = render(<Pollutant />);
    expect(tree).toMatchSnapshot();
  });
});
