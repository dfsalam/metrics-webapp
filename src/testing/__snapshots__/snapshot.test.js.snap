// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render correctly Banner 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="banner-container"
      >
        <div
          class="banner-text"
        >
          <h2 />
          <h3 />
          <h3 />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="banner-container"
    >
      <div
        class="banner-text"
      >
        <h2 />
        <h3 />
        <h3 />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components render correctly Detail 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3>
        Is loading...
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3>
      Is loading...
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components render correctly Pollutant 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pol-container"
      >
        <h3
          class="pol-text"
        />
        <h3
          class="pol-text"
        >
          undefined μg/m
          <sup>
            3
          </sup>
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pol-container"
    >
      <h3
        class="pol-text"
      />
      <h3
        class="pol-text"
      >
        undefined μg/m
        <sup>
          3
        </sup>
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components render correctly Search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        id="searchInput"
        placeholder="Search..."
        type="text"
      />
    </div>
  </body>,
  "container": <div>
    <input
      id="searchInput"
      placeholder="Search..."
      type="text"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components render correctly TownContainer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="cntTitle"
        >
          STATS BY TOWN
        </h2>
        <div
          class="cnt-body"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="cntTitle"
      >
        STATS BY TOWN
      </h2>
      <div
        class="cnt-body"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
